audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/model/AuditMessage.java

```diff
@@ -39,17 +39,29 @@
 @JsonAutoDetect(fieldVisibility = JsonAutoDetect.Visibility.ANY)
 public class AuditMessage {
```

Comment:
Pretty generic change to class.  We go this route, this change should also be done to the other AuditMessage classes.  Would suggest we raise a ticket to define and implement the standard for all AuditMessage, AuditErrorMessage, and AuditSuccessMessage.

Reply:
If we do raise a ticket i would prefer to use Immutables which would at on the very least add Optional support, really good builders (ours are awful) automatic json support, create new instance based upon another etc etc.

Reply:
I've done a test with Immutables this evening out of curiosity and it works and remove a large amount of code and adds lots of flexibility.